src/3d/qgs3daxis.cpp

```diff
@@ -175,64 +213,41 @@ void Qgs3DAxis::onTouchedByRay( const Qt3DRender::QAbstractRayCaster::Hits &hits
   }
   else if ( mLastClickedButton == Qt::MouseButton::LeftButton ) // handle cube face clicks
   {
-    if ( mMenu->isVisible() )
-      hideMenu();
+    hideMenu();
 
     if ( mHitsFound != -1 )
     {
       if ( hits.at( mHitsFound ).entity() == mCubeRoot || hits.at( mHitsFound ).entity()->parent() == mCubeRoot )
       {
-#ifdef DEBUG
-        switch ( hits.at( mHitsFound ).primitiveIndex() / 2 )
-        {
-          case 0:
-            qDebug() << "East face";
-            break;
-
-          case 1:
-            qDebug() << "West face ";
-            break;
-
-          case 2:
-            qDebug() << "North face ";
-            break;
-
-          case 3:
-            qDebug() << "South face";
-            break;
-
-          case 4:
-            qDebug() << "Top face ";
-            break;
-
-          case 5:
-            qDebug() << "Bottom face ";
-            break;
-        }
-#endif
         switch ( hits.at( mHitsFound ).primitiveIndex() / 2 )
         {
           case 0: // "East face";
+            QgsTraceMsg( "Qgs3DAxis: East face clicked" );
```

Comment:
Why are you using trace message here instead of QgsDebugMsgLevel?

Reply:
Because, IMHO, this is more a message to be used when debugging and not really useful to the end user.

Reply:
That's exactly what QgsDebugMsgLevel/QgsDebugMsg is for. (It's not the same as QgsLogger.)

Reply:
I added an new define in https://github.com/qgis/QGIS/blob/cbf291023535b23f85c5442755e6f79395ea8f30/src/core/qgslogger.h#L34 to handle this.

Reply:
That's just a less efficient version of QgsDebugMsgLevel 😝 . QgsDebugMsgLevel has logic to avoid the cost of string construction when the current debug level means that the message will be filtered.

Reply:
@nyalldawson I pushed a new version. Do you prefer?